compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -447,6 +449,14 @@ public void visit(BLangFunction funcNode, AnalyzerData data) {
             validateAnnotationAttachmentCount(funcNode.returnTypeAnnAttachments);
             ((BInvokableTypeSymbol) funcNode.symbol.type.tsymbol).returnTypeAnnots.addAll(
                     getAnnotationAttachmentSymbols(funcNode.returnTypeAnnAttachments));
+            if (returnTypeNode.getKind() == TUPLE_TYPE_NODE) {
+                for (BLangMemberTypeNode memberTypeNode : ((BLangTupleTypeNode) returnTypeNode).memberTypeNodes) {
```

Comment:
Isn't it just enough to do `visit((BLangTupleTypeNode) returnTypeNode, data)` here?
There is also a difference here and the visit where we don't analyse the tuples `restParamType`

Reply:
But Ideally, I think we should have visited the return type, whether it is a tuple or not. Shall we add more test cases for this scenario?